types/externs.d.ts

```diff
@@ -6,7 +6,25 @@
 
 import _Crdp from 'devtools-protocol/types/protocol';
 import _CrdpMappings from 'devtools-protocol/types/protocol-mapping';
-import {ParseSelector} from 'typed-query-selector/parser';
+import {ParseSelectorToTagNames} from 'typed-query-selector/parser';
+
+/** Merge properties of the types in union `T`. Where properties overlap, property types becomes the union of the two (or more) possible types. */
+type MergeTypes<T> = {
+  [K in (T extends unknown ? keyof T : never)]: T extends Record<K, infer U> ? U : never;
+};
+
+// Helper types for strict querySelector/querySelectorAll that includes the overlap
+// between HTML and SVG node names (<a>, <script>, etc).
+// see https://github.com/GoogleChrome/lighthouse/issues/12011
+type HtmlAndSvgElementTagNameMap = MergeTypes<HTMLElementTagNameMap|SVGElementTagNameMap> & {
+  // Fall back to Element (base of HTMLElement and SVGElement) if no specific tag name matches.
+  [id: string]: Element;
+};
+type QuerySelectorParse<I extends string> = ParseSelectorToTagNames<I> extends infer TagNames ?
+  TagNames extends Array<string> ?
+    HtmlAndSvgElementTagNameMap[TagNames[number]] :
+    Element: // Fall back for queries typed-query-selector fails to parse, e.g. `'[alt], [aria-label]'`.
```

Comment:
`typed-query-selector` returns `unknown` from `ParseSelectorToTagNames` for both situations it can't handle _and_ known error states (like `document.querySelector('')`). As a result, with `Element` here it'll still give that type for those bad queries instead of `never`, but it has to be included for queries like `'[alt], [aria-label]'` (from [`page-functions`](https://github.com/GoogleChrome/lighthouse/blob/20c8080d28507affed505c55c1b47179a8ccc648/lighthouse-core/lib/page-functions.js#L463)) which need the `Element` fall back.

Not much to do to get around this except write our own version, but no worse than the current state of things or the default typescript `querySelector` types :)